docs/rfc/rfc-1.adoc

```diff
@@ -503,37 +503,58 @@ this transaction.
 
 [[transaction-incentives]]
 === Transaction Incentives
-
-Governable Parameters:
-
-- `max_gas_refund_price`: The highest amount of gwei that the gas refund
-  contract will pay out per gas for a refund transaction.
-
-There are 7 main transactions that require incentives:
-
-==== Transactions That Must Be Submitted By Operators
-- Submit distributed key generation results for new wallet creation (weekly).
-- Signal that funds need to be moved between wallets if a <<heartbeat,heartbeat>> fails.
-- Signal that a staker is not performing their duty and should be marked as
-ineligible for rewards (as in the case that they were inactive during
-heartbeats).
-
-==== Transactions That Can Be Submitted By Anyone
-- Begin distributed key generation for new wallet creation (weekly).
-- Provide the proof for a <<sweeping,sweep>> transaction.
-- Provide the proof for a <<redemption,redemption>> transaction.
-- Provide the proof that funds have moved between wallets.
-
-To make sure that these transactions get posted, we need the DAO to fund an ETH
-pool that provides gas refunds for anyone submitting these transactions. We
-know the amount of gas spent, and we know the gas price used, and so can
-reimburse the operator.
-
-To prevent abuse, we need to let governance set caps (`max_gas_refund_price`)
-on how much eth can be reimbursed (so that a miner can't drain the pool).
-
-Further ideas are to run bots like those provided by https://www.gelato.network/[Gelato]
-in order to prevent problems like operators front-running each other.
+Transaction incentives are more deeply explored in link:rfc-6.adoc[RFC 6:
+Transaction Incentives]. Summarized:
+
+There are three different types of transactions: <<operator-only,Operator-Only>>,
+<<public-knowledge,Public-Knowledge>>, and <<punishment,Punishment>>.
+
+[[operator-only]]
+==== Operator-Only
+Operator-Only transactions are where only the operators have access to the
+information required to assemble the transaction with the right input
+parameters.
+
+In order to avoid all operators racing to submit the
+transaction at the same time, we have an off-chain gentleman's agreement to
+submit based on the operator's position in the group (can use the hash of the
+group's pubkey).
+
+If the designated operator does not submit their transaction before a timeout
+expires, the duty moves to the next operator and the group can sign a
+transaction to mark that operator as inactive. Such a transaction would be a
+<<punishment,Punishment>>.
```

Comment:
I just want to make sure I understand correctly (plus have one question):
Let's say we're talking about transaction submitting DKG result. So in the case where the first operator does not submit the DKG result before the timeout, second operator would send that transaction. This transaction with the DKG result sent by operator_2 would not report that operator_1 was inactive during result submission. It would be a transaction of Operator-Only type. It would be reimbursed (maybe during approval and not during submission, but still).
To report the inactivity of operator_1 and it's ineligibility for rewards a separate transaction would need to be sent by a willing party. That would be transaction of the Punishment type. This transaction would not be reimbursable.
Is that right so far?
If so, what would be the incentive here for posting of this second transaction? We're not slashing here, and funds saved by not paying the rewards to operator_1 will not increase the rewards for other operators, so why would anybody care to post that?

Reply:
Great catch!

Your understanding is totally correct - operator_1 failed, so the duty falls to operator_2 and the wallet can sign a transaction to disable rewards for operator_1.

As you say, there's no incentive to post that transaction, and only the operators have the required info, so even though it results in a punishment we need to incentivize it like it's a Operator-Only transaction (refund the eth). I'll amend the RFC 😇

Reply:
addressed in [cf33272](https://github.com/keep-network/tbtc-v2/pull/125/commits/cf33272c245caa20cdbe6e38d22fe80085d790c3)